src/contexts/pageContext.tsx

```diff
@@ -0,0 +1,39 @@
+import { createContext, useState, useEffect } from 'react'
+import { ProviderProps } from '../types/contexts'
+import { type FlashProps } from '../types/flashMessages'
+
+interface PageContextType {
+  flashProps: FlashProps
+  setFlashProps: (props: FlashProps) => void
+}
+
+const defaultFlashProps: FlashProps = {
+  type: 'info',
+  message: '',
+  hidden: true,
+}
+
+const PageContext = createContext<PageContextType>({
+  flashProps: defaultFlashProps,
+  setFlashProps(this: PageContextType, props: FlashProps) {
+    this.flashProps = props
+  },
+})
+
+const PageProvider = ({ children }: ProviderProps) => {
+  const [flashProps, setFlashProps] = useState<FlashProps>(defaultFlashProps)
+
+  useEffect(() => {
+    if (flashProps.hidden == false) {
+      setTimeout(() => setFlashProps({ ...flashProps, hidden: true }), 3000)
```

Comment:
3 seconds might be a short window of visibility for longer error messages. As we continue to build out the app, we'll see if this needs to be extended. If it lasts too long users will think there's no way to get rid of it.

Reply:
We could also have just set them to `defaultFlashProps` again, but frankly this works and it's what I thought of first, so I don't see a reason for that.